### ISSUE TITLE
Using a signal in a ProtectedRoute condition panics

### ISSUE THREAD

**Issue by sambonbonne** [leptos-rs/leptos#1584]
**Describe the bug**

I try to use a signal in a `ProtectedRoute` condition function but it makes `cargo leptos watch` display a panic with the error `runtime to be alive: ()`.

**Leptos Dependencies**

```toml
leptos = { version = "0.4", features = ["nightly"] }
leptos_axum = { version = "0.4", optional = true }
leptos_meta = { version = "0.4", features = ["nightly"] }
leptos_router = { version = "0.4", features = ["nightly"] }
```

**To Reproduce**

I made a reproduction repository: [sambonbonne/leptos-protected-route-reproduction](https://github.com/sambonbonne/leptos-protected-route-reproduction)

If you want to reproduce yourself, init a new Leptos project with `cargo leptos new --git leptos-rs/start-axum` and add this in the `App` component (and create dumbs `Authenticated` and `Unauthenticated` components):

```rust
let (is_authenticated, set_is_authenticated) = create_signal(cx, false); // may change to true if the use logs in

view! { cx,
  <Router>
    <main>
      <Router>
        <ProtectedRoute path="/authenticated" view=AuthenticatedPage redirect_path="/not-authenticated" condition=move |_| is_authenticated.with(|auth| auth == true) />
        <ProtectedRoute path="/not-authenticated" view=UnauthenticatedPage redirect_path="/authenticated" condition=move |_| is_authenticated.with(|auth| auth == false) />
      </Router>
    </main>
  </Router>
}
```

Run `cargo leptos watch` and wait for it to be ready.

Then, try to open [localhost:3000/authenticated](http://localhost:3000/authenticated) and [localhost:3000/not-authenticated](http://localhost:3000/not-authenticated) in your browser

**Expected behavior**

I expect the protected route to load or redirect properly, depending on the signal status.

But the page does not load. In the `carog leptos watch` logs, I see the following logs:

```log
thread '<unnamed>' panicked at src/app.rs:18:52:
runtime to be alive: ()
thread 'tokio-runtime-worker' panicked at /home/sam/.cargo/registry/src/index.crates.io-6f17d22bba15001f/leptos_axum-0.4.10/src/lib.rs:657:44:
called `Option::unwrap()` on a `None` value
```

Note that the line shown in the logs for `src/app.rs` is the line of the definition of the function, not the line of the call: if you defined your condition function with a `let guard = move |_| …` and use `condition=guard`, the line number of the panick will be the line of `let guard = …`.

**Additional context**

I'm a new Leptos user, I found the `ProtectedRoute` by looking at the Leptos Router documentation but I did not find if the condition could contain signals so I don't know if this is really a bug.

**Comment by gbj** [comment]
At first glance it looks to me like this is the same issue as #1397. It should be fixed on the main branch or on 0.5.0. I'd recommend following the migrations steps in the [0.5.0-beta release notes](https://github.com/leptos-rs/leptos/releases/tag/v0.5.0-beta) and seeing if it works.

Sorry for the trouble!

**Comment by sambonbonne** [comment]
It works well with `0.5.0-beta`! Thank you, I should have try it before!

No problem for the trouble, I was not blocked by this bug. Thanks for your help!
